source/operations/cert-manager/cert-manager-tenants.rst

```diff
@@ -0,0 +1,252 @@
+
+     .. code-block:: shell
+        :class: copyable
+
+        kubectl get configmap coredns -n kube-system -o jsonpath="{.data}"
+
+     Different Kubernetes providers manage the root domain differently.
+     Check with your Kubernetes provider for more information.
+
+   - ``tenant-name`` is the name provided to your tenant in the ``metadata.name`` of the Tenant YAML. 
+     For this example it is ``myminio``.
+
+   - ``namespace`` is the value created earlier where the tenant will be installed.
+     In the tenant YAML, it is defined in the the ``metadata.namespace`` field. 
+     For this example it is ``tenant-1``.
+
+1. Request a ``Certificate`` for the domains mentioned above
+
+   Create a file called ``tenant-1-minio-certificate.yaml`` with the following contents: 
+
+   .. code-block:: yaml
+      :class: copyable
+
+      # tenant-1-minio-certificate.yaml
+      apiVersion: cert-manager.io/v1
+      kind: Certificate
+      metadata:
+        name: tenant-certmanager-cert
+        namespace: tenant-1
+      spec:
+        dnsNames:
+          - "minio.tenant-1"
+          - "minio.tenant-1.svc"
+          - 'minio.tenant-1.svc.cluster.local'
+          - '*.minio.tenant-1.svc.cluster.local'
+          - '*.myminio-hl.tenant-1.svc.cluster.local'
+          - '*.myminio.minio.tenant-1.svc.cluster.local'
+        secretName: myminio-tls
+        issuerRef:
+          name: tenant-1-ca-issuer
+
+   .. tip::
+
+      For this example, the Tenant name is ``myminio``. 
+      We recommend naming the secret in the field ``spec.secretName`` as ``<tenant-name>-tls``, following the naming convention the MinIO Operator uses when creating certificates without cert-manager.
+
+2. Apply the certificate resource to the cluster.
+
+   .. code-block:: shell
+      :class: copyable
+
+      kubectl apply -f tenant-1-minio-certificate.yaml
+
+4) Configure the tenant to use the certificate
+----------------------------------------------
+
+To use cert-manager, the tenant spec must:
+
+- Disable AutoCert by setting the ``spec.requestAutoCert`` field to ``false``. 
+
+  This instructs the MinIO Operator to not attempt to issue certificates and instead rely on cert-manager to provide them in a secret.
+- Reference the Secret containing the TLS certificate from the previous procedure in `spec.externalCertSecret`.
+
+Modify the tenant YAML ``spec`` section to reflect the above requirements.
+   
+.. code-block:: yaml
+   :emphasize-lines: 6,9,11
+
+   apiVersion: minio.min.io/v2
+   kind: Tenant
+   metadata:
+     name: myminio
+     namespace: tenant-1
+   spec:
+   ...
+     ## Disable default tls certificates.
+     requestAutoCert: false
+     ## Use certificates generated by cert-manager.
+     externalCertSecret:
+       - name: myminio-tls
+         type: cert-manager.io/v1
+   ...
+
+5) Trust the tenant's CA in MinIO Operator
+------------------------------------------
+
+MinIO Operator can trust as many CA certificates as provided. 
```

Comment:
This is slightly out of left-field in terms of relevance.

I think the key here is:

1) The Operator cannot trust the cert-manager issued CA by default (this seems weird if we are all using the same ClusterIssuer but whatever)

2) We must pass the Tenant CA to the Operator

3) This requires creating a secret in the `minio-operator` namespace that references the Tenant CA created in this procedure.

Reply:
Fair. Updated to reflect.